### PR TITLE
[CBRD-23920] Implement APIs for CDC

### PR DESCRIPTION
**Pull request by hornetmj**
http://jira.cubrid.org/browse/CBRD-23920

commit: [a03f640](https://github.com/CUBRID/cubrid/pull/2710/commits/a03f640955f9f260b631326bc6e7da2a72ad2df8)
- create cubrid_log.c/h files in src/api directory to implement CDC interfaces
- the libcubridcs.so library includes them
- the cubrid_log.h file exists in $CUBRID/include directory after cubrid installation

### CODE REVIEW

**Comment by hornetmj** [comment]
@JoohoK
commit: [c026d3c](https://github.com/CUBRID/cubrid/pull/2710/commits/c026d3c595b9ca6e3b03b3439f9cb028948f1ee7)
- declaration of API functions
- declaration of API struct
- define error codes
- make skeleton code for these functions
- error check for parameters of these functions
- remove the '#if defined(CS_MODE)' from the header file because it will be open to the user

**Comment by hornetmj** [comment]
@JoohoK
commit: [4188f8c](https://github.com/CUBRID/cubrid/pull/2710/commits/4188f8c2021508bd3ec73f32fea155b9b4fb1289)
- connect/disconnect to the server using css_* functions
  - this will be replaced to the low-level functions after analysis

**Comment by hornetmj** [comment]
@JoohoK
commit: [7d91e53](https://github.com/CUBRID/cubrid/pull/2710/commits/7d91e53ec3def550f386246cf52d0007d5f4e5a1)
- implement the cubrid_log_set_* function series
- check if an API function can be called at the stage and transition to the next stage if necessary
- define extra error codes
  - CUBRID_LOG_INVALID_FUNC_CALL_STAGE
  - CUBRID_LOG_FAILED_MALLOC

please check the name of error codes added newly. If you think they need to rename, then leave some feedback.

**Comment by hornetmj** [comment]
@JoohoK
commit: [83cbd09](https://github.com/CUBRID/cubrid/pull/2710/commits/83cbd09bc1ffc67e20bcc24fc42d1f379e778f6c)
- connect/disconnect to the server using css_common_connect() and css_free_conn()
- manual test
  - cubrid_log_connect_server()
    - confirm that a new client request is received by using the css_process_new_client() function in the cub_server using gdb
  - cubrid_log_finalize()

**Comment by hornetmj** [comment]
@JoohoK
commit: [b9c1c61](https://github.com/CUBRID/cubrid/pull/2710/commits/b9c1c6147b8d9813fbe17cb2fc9ffc8daf620fd7)
- send properties set by the user to the server
  - exclude variable length properties like extraction users and extraction tables
- manual test
  - cubrid_log_set_* ()
    - print out the values of properties received to the server error log

File: src/api/cubrid_log.c [hunk 1]
```diff
@@ -0,0 +1,1317 @@
...
+
+  CSS_QUEUE_ENTRY *queue_entry;
+  int err_code;
+
+  ptr = or_pack_int (request, g_max_log_item);
+  ptr = or_pack_int (ptr, g_extraction_timeout);
+  ptr = or_pack_int (ptr, g_all_in_cond);
+  ptr = or_pack_int (ptr, g_extraction_user_count);
+  ptr = or_pack_int (ptr, g_extraction_table_count);
+
```
Comment (joohok) [inline src/api/cubrid_log.c:418]: I think 'g_extraction_user' and 'g_extraction_table' are required to be packed also

File: src/api/cubrid_log.c [hunk 2]
```diff
@@ -0,0 +1,1317 @@
...
+  ptr = or_unpack_int (recv_data, &reply_code);
+
+  if (reply_code != NO_ERROR)
+    {
+      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_LSA_NOT_FOUND);
+    }
+
+  or_unpack_log_lsa (ptr, lsa);
+
```
Comment (joohok) [inline src/api/cubrid_log.c:557]: In server side, lsa will be converted to uint64_t and packed with or_pack_int64()

File: src/api/cubrid_log.c [hunk 3]
```diff
@@ -0,0 +1,1317 @@
...
+  if (reply_code != NO_ERROR)
+    {
+      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_INVALID_LSA);
+    }
+
+#if !defined (NDEBUG)
+  printf ("send_next_lsa = %lld|%d\n", LSA_AS_ARGS (next_lsa));
+#endif
+
+  ptr = or_unpack_log_lsa (ptr, next_lsa);
```
Comment (joohok) [inline src/api/cubrid_log.c:677]: lsa will be sent as a uint64_t

**Pull request merged**